### DISCUSSION TITLE
Support for LinuxRT

### DISCUSSION THREAD

**Discussion by citlaligm** [JKISoftware/Caraya#165] [Q&A]
Hello, 
I saw in a colleague's presentation that this tool is not support in Real-Time targets (NI specifically). I also tried to run a dummy test under a RT target. It ran without errors but I was unable to see the results windows (guessing a UI limitation) is there a workaround, maybe a text file report or something? Is there any plans to support this? 

My use case is to test code that usually runs under a RT target (PPL) but I'd like to build unit test cases. 

Thank you!

**Comment by francois-normandin** [comment]
Hi @citlaligm , 

Caraya is actually supported on RT targets, but as you pointed out, there are no User interfaces (on most of them) so Caraya suppresses that code when running on those targets. What is not supported are automated test discovery such as the ones from that call VI Server for execution.

I suggest you open up the Tree View from the palette, where support for RT and non-RT targets is clearly labeled.
![image](https://github.com/JKISoftware/Caraya/assets/11728548/ea8c2794-feeb-49b4-b0c2-05281a7fd17a)

The advanced Tree View goes in a bit more detail.
![image](https://github.com/JKISoftware/Caraya/assets/11728548/1fd6b52d-d886-4658-8c41-834f11df2180)


**The simplest way to get your Test Results on RT** is to create a Test Suite and inject a Test Report class.
Caraya is pre-loaded with support for JUnit and text-based report, just like the ones you can export from the Test Manager UI's menus.
Check out the wiki if you want to create your own custom reports: https://github.com/JKISoftware/Caraya/wiki/Creating-Custom-Test-Reports

You can see the example code below, where all your tests have been wrapped into a single high-level VI:
![image](https://github.com/JKISoftware/Caraya/assets/11728548/077adcb1-cf10-4c6f-939d-705252f6dfcc)

But there is more. You can also use the Test Suite and wait for completion, then use the Test Result class output to essentially report through your favorite mechanism.

![image](https://github.com/JKISoftware/Caraya/assets/11728548/e0ca7549-f297-4604-9d86-39d9e358280a)

I hope this helps!

PS.
_Stay tuned, because version 1.4.0 will actually extend the reporting strategy even more. I've got it working, but I still am performing tests to ensure it is fully backwards-compatible._

**Comment by citlaligm** [comment]
@francois-normandin thank you for your comprehensive answer. This is exactly what I need.
